catalog/app/containers/Bucket/requests.js

```diff
@@ -269,18 +269,61 @@ export const bucketStats = async ({ req, s3, bucket, overviewUrl }) => {
   throw new Error('Stats unavailable')
 }
 
-export const metadataSchema = async ({ s3, bucket, schemaUrl }) => {
+export class ErrorFileNotFound extends Error {}
+
+export class ErrorVersionNotFound extends Error {}
+
+export const latestFile = async ({ s3, bucket, path }) => {
+  const doesExists = await ensureObjectIsPresent({
+    s3,
+    bucket,
+    key: path,
+  })
+  if (!doesExists) {
+    throw new ErrorFileNotFound(`${path} for ${bucket} does not exists`)
+  }
+
+  const versions = await objectVersions({
+    s3,
+    bucket,
+    path,
+  })
+  const { id } = R.find(R.prop('isLatest'), versions)
+  const version = id === 'null' ? undefined : id
+
+  const doesVersionExists = await ensureObjectIsPresent({
+    s3,
+    bucket,
+    key: path,
+    version,
+  })
+  if (!doesVersionExists) {
+    throw new ErrorVersionNotFound(
+      `${path} for ${bucket} and version №${version} does not exists`,
+    )
+  }
+
+  return s3
+    .getObject({
+      Bucket: bucket,
+      Key: path,
+      VersionId: version,
+    })
+    .promise()
+}
+
+export const metadataSchema = async ({ s3, schemaUrl }) => {
   if (!schemaUrl) return null
 
+  const { bucket, key } = s3paths.parseS3Url(schemaUrl)
```

Comment:
i think maybe we should expect `version` here as well (@sir-sigurd do we support versioned urls for schemas?)

Reply:
like, use the version if provided or fetch latest if not